torch/testing/_internal/dynamo_test_failures.py

```diff
@@ -37,6 +16,48 @@ def find_test_dir():
 # *These are not exactly unittest.expectedFailure and unittest.skip. We'll
 # always execute the test and then suppress the signal, if necessary.
 # If your tests crashes, or is slow, please use @skipIfTorchDynamo instead.
+#
+# The expected failure and skip files are located in test/dynamo_skips and
+# test/dynamo_expected_failures. They're individual files rather than a list so
+# git will merge changes easier.
+
+
+def find_test_dir():
+    # Find the path to the dynamo expected failure and skip files.
+    from os.path import abspath, basename, dirname, exists, join, normpath
+
+    if sys.platform == "win32":
+        return None
+
+    # Check relative to this file (local build):
+    test_dir = normpath(join(dirname(abspath(__file__)), "../../../test"))
+    if exists(join(test_dir, "dynamo_expected_failures")):
+        return test_dir
+
+    # Check relative to __main__ (installed builds relative to test file):
+    main = sys.modules["__main__"]
+    file = getattr(main, "__file__", None)
+    if file is None:
+        # Generated files do not have a module.__file__
+        return None
+    test_dir = dirname(abspath(file))
+    while dirname(test_dir) != test_dir:
+        if basename(test_dir) == "test" and exists(
+            join(test_dir, "dynamo_expected_failures")
+        ):
+            return test_dir
+        test_dir = dirname(test_dir)
+
+    # Not found
+    return None
+
+
+test_dir = find_test_dir()
+if not test_dir:
+    logger = logging.getLogger(__name__)
```

Comment:
what logger does this go to?

Can you test running
```
PYTORCH_TEST_WITH_DYNAMO=1 python test/test_testing.py -k test_filtering_env_var
```
I hope this does not break that test that reads logs

Reply:
That passes for me. It goes to the "standard" logger, I guess? It only logs in the case where it couldn't find the skip.